src/librustdoc/visit_ast.rs

```diff
@@ -78,7 +78,6 @@ impl<'a, 'tcx> RustdocVisitor<'a, 'tcx> {
             &krate.item,
```

Comment:
This is unrelated, but I just noticed that two lines above we use `DUMMY_SP` as the crate's span. Do you know why we don't just use `krate.item.inner` (which I think is the whole span of the crate)?

Reply:
Probably just an oversight, I can make a follow-up fixing it.

Reply:
Hmm, strangely this gives lifetime errors:
```
error[E0515]: cannot return value referencing temporary value
   --> src/librustdoc/visit_ast.rs:128:9
    |
77  |             &Spanned { span, node: hir::VisibilityKind::Public },
    |              --------------------------------------------------- temporary value created here
...
128 |         top_level_module
    |         ^^^^^^^^^^^^^^^^ returns a value referencing data owned by the current function
```
I don't know why, Span doesn't have a lifetime.

Reply:
Ugh, as usual people are adding too many lifetimes and some of them are wrong. Here's a fix:
```diff
diff --git a/src/librustdoc/visit_ast.rs b/src/librustdoc/visit_ast.rs
index ca30d8f0d46..d852728b749 100644
--- a/src/librustdoc/visit_ast.rs
+++ b/src/librustdoc/visit_ast.rs
@@ -71,9 +71,10 @@ fn store_path(&mut self, did: DefId) {
     }
 
     crate fn visit(mut self, krate: &'tcx hir::Crate<'_>) -> Module<'tcx> {
+        let span = krate.item.inner;
         let mut top_level_module = self.visit_mod_contents(
-            krate.item.inner,
-            &Spanned { span: rustc_span::DUMMY_SP, node: hir::VisibilityKind::Public },
+            span,
+            &Spanned { span, node: hir::VisibilityKind::Public },
             hir::CRATE_HIR_ID,
             &krate.item,
             self.cx.tcx.crate_name,
@@ -130,7 +131,7 @@ fn store_path(&mut self, did: DefId) {
     fn visit_mod_contents(
         &mut self,
         span: Span,
-        vis: &'tcx hir::Visibility<'_>,
+        vis: &hir::Visibility<'_>,
         id: hir::HirId,
         m: &'tcx hir::Mod<'tcx>,
         name: Symbol,
```